### ISSUE TITLE
Needs to support hot reloading when running in Docker container

### ISSUE THREAD

**Issue by nbeck415** [nbeck415/bun-nextjs-dockerized#1]

**Comment by Sasha95** [comment]
docker-compose.yml

```
version: '3.8'

services:
  next:
    labels:
      shipyard.route: '/'
    build:
      context: .
    ports:
      - '3000:3000'
    volumes:
      - ./public:/app/public
      - ./src:/app/src
    develop:
      watch:
        - action: sync
          path: ./public
          target: /app/public
        - action: sync
          path: ./src
          target: /app/src
        - action: rebuild
          path: package.json
```

**Comment by nbeck415** [comment]
@Sasha95 this maintains the current behavior.

The question is that Bun's [hot reloading](https://bun.sh/docs/runtime/hot#hot-mode) functionality is a soft reload, which is slightly different (and lighter/more responsive) from what Compose Watch does. I opened this issue as a `TODO` to look into the most optimal way to hot reload a Bun app, since I like Bun's inbuilt method. 

Currently exploring this:
1. Using a Docker build arg to allow user to add flags to their Docker start `CMD` (for using with or without Bun hot reload)
2. Studying this behavior; looking into whether it makes sense to also use `compose watch` or if that becomes redundant

**TLDR:** deciding between Bun hot reload and Compose Watch, or some combo
